File: week_07/day_1/Try_Catch_Exceptions/code/CheckPoint_1_custom_exceptions/app/src/main/java/example/codeclan/com/checkpoint_1_custom_exceptions/PetShop.java
package example.codeclan.com.checkpoint_1_custom_exceptions;

/**
 * Created by user on 30/08/2017.
 */

import java.util.ArrayList;

public class PetShop{

    private ArrayList<Pet> stock;

    public PetShop(){
        stock = new ArrayList<Pet>();
    }

    public Pet findPetByName(String searchName) {
        // add an if statement to catch a null String
        if (searchName == null){
            // and throw a new exception

                throw new NullStringException("Cannot search for a pet with null instead of a name String");

        }
        String searchLower = searchName.toLowerCase();
        for (Pet pet : stock){
            String petName = pet.getName().toLowerCase();
            if ( petName.equals(searchLower) ) return pet;
        }
        return null;
    }

    public void addPet(Pet pet){
        stock.add(pet);
    }

}

File: week_10/c_sharp_course/week_1/day_5/try_catch/code/end_code/PetShop.cs
using System.Collections.Generic;

public class PetShop
{
  private List<Pet> stock;

  public PetShop()
  {
    stock = new List<Pet>();
  }

  public Pet FindPetByName(string searchName)
  {    
    if (searchName == null)
    {
      throw new NullStringException("Cannot search for a pet with null instead of a name string");
    }

    string searchLower = searchName.ToLower();
    foreach (var pet in stock)
    {
      string petName = pet.GetName().ToLower();
      if ( petName.Equals(searchLower) ) 
      {
        return pet;
      }
    }
    return null;
  }

  public void AddPet(Pet pet)
  {
    stock.Add(pet);
  }

}